server/api-manager.js

```diff
@@ -974,19 +1001,28 @@ class MindbodyQueries {
         .then((value) => {
           //console.log(Object.keys(value));
           if (!value.PaginationResponse) {
+            if (request.verb === "GET") {
+              this.cachedResults[hashCode] = {
+                siteId: request.siteId,
+                url: request.url,
+                creationTime: Date.now(),
+                result: value,
+              };
+              //console.log("MindbodyAccess stored this cacheTuple " + JSON.stringify(this.cachedResults[hashCode]) + " at hash value: " + hashCode);
+            }
             return Promise.resolve(value);
           }
           let totalResults = value.PaginationResponse.TotalResults;
           let requestedLimit = value.PaginationResponse.RequestedLimit;
           let resultsSeenSoFar = requestedLimit;
-          let url = request.url;
+          let initialUrl = request.url;
           while (resultsSeenSoFar < totalResults) {
             this.requestNum++;
             if (this.atLimit()) {
               return Promise.reject(Error("Mindbody request limit reached"));
             }
             const resultsPerPage = 200;
-            request.url = url + "&limit=" + resultsPerPage + "&offset=" + resultsSeenSoFar;
+            request.url = initialUrl + "&limit=" + resultsPerPage + "&offset=" + resultsSeenSoFar;
```

Comment:
I renamed url to initialUrl because I was referencing it and this line was too confusing for me to read at first.